pjlib/src/pj/atomic_queue.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (C) 2024 Teluu Inc. (http://www.teluu.com)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#include <pj/assert.h>
+#include <pj/atomic_queue.h>
+#include <pj/errno.h>
+#include <pj/pool.h>
+#include <pj/string.h>
+#include <atomic>
+
+class AtomicQueue {
+public:
+    /**
+     * Constructor
+     */
+    AtomicQueue(unsigned maxItemCnt, unsigned itemSize, const char* name = ""):
+                maxItemCnt_(maxItemCnt),
+                itemSize_(itemSize),
+                ptrWrite(0),
+                ptrRead(0),
+                buffer(NULL),
+                name_(name)
+    {
+        buffer = new char[maxItemCnt_ * itemSize_];
+
+        /* Surpress warning when debugging log is disabled */
+        PJ_UNUSED_ARG(name_);
+
+        // PJ_LOG(5, (name_, "Created AtomicQueue: maxItemCnt=%d itemSize=%d",
```

Comment:
Rather than comment it, better disable it as @nanangizz suggested, such as using macro TRACE, so we can easily enable it when logging is needed.